src/main/java/bio/terra/service/auth/iam/IamProviderInterface.java

```diff
@@ -217,4 +217,17 @@ String getPetToken(AuthenticatedUserRequest userReq, List<String> scopes)
    *     statuses
    */
   RepositoryStatusModelSystems samStatus();
+
+  /**
+   * @param accessToken valid oauth token for the account fetching the group email
+   * @param groupName name of Firecloud managed group
+   * @return the group's email address
+   */
+  String getGroupEmail(String accessToken, String groupName) throws InterruptedException;
+
+  /**
+   * @param accessToken valid oauth token for the account creating the group
+   * @param groupName name of Firecloud managed group to create
+   */
+  void createGroup(String accessToken, String groupName) throws InterruptedException;
```

Comment:
bah, nevermind about the return.  I see that the API doesn't return anything.  Bummer.  Do we need a delete too?  Not sure if we ever plan on implementing that if all instances of a DUOS ID are gone

Reply:
Good questions!

- Void method: yeah, I wish the SAM API behaved differently in this case!  I toyed with making `createGroup` return the group's email though.  In fact, I think I'll make that change to reduce the burden on the caller, as it's clearly not the most intuitive.
- Delete group: I ended up adding this in a later commit!  The way it is presently used is if we successfully create a group but fail to insert it in the DB, we delete it to avoid proliferation of unrecorded groups. https://github.com/DataBiosphere/jade-data-repo/blob/7fcaf145670773d610a26ed56d9cf0db03ccba62/src/main/java/bio/terra/service/duos/DuosService.java#L47-L52

Reply:
OMG, giant facepalm...I forgot that I was reviewing commit-by-commit.